api/src/main/java/mb/pie/api/Pie.java

```diff
@@ -118,4 +119,10 @@
      * @throws RuntimeException when closing fails.
      */
     void close();
+
+    TaskDefs getTaskDefs();
+
+    Callbacks getCallbacks();
```

Comment:
I don't like that these internals are exposed through the `Pie` interface. Maybe a better way would be to add `void addToChildBuilder(PieChildBuilder builder)`, which `PieImpl` can then implement to add itself to the child builder. This at least makes the intent clear and prevents abuse of these exposed internals.